src/lib.rs

```diff
@@ -356,7 +352,69 @@ mod tests {
         simulation.run().unwrap()
     }
 
-    fn simulate_dkr(keys: &mut Vec<LocalKey>) {
+    fn simulate_dkr_removal(keys: &mut Vec<LocalKey>, remove_party_indexes: Vec<usize>) {
+        let mut broadcast_messages: HashMap<usize, Vec<RefreshMessage<GE>>> = HashMap::new();
+        let mut new_dks: HashMap<usize, DecryptionKey> = HashMap::new();
+        let mut refresh_messages: Vec<RefreshMessage<GE>> = Vec::new();
+        let mut party_key: HashMap<usize, LocalKey> = HashMap::new();
+
+        for key in keys.iter_mut() {
+            let (refresh_message, new_dk) = RefreshMessage::distribute(key);
+            refresh_messages.push(refresh_message.clone());
+            new_dks.insert(refresh_message.party_index, new_dk);
+            party_key.insert(refresh_message.party_index, key.clone());
+        }
+
+        for refresh_message in refresh_messages.iter() {
+            broadcast_messages.insert(refresh_message.party_index, Vec::new());
+        }
+
+        for refresh_message in refresh_messages.iter_mut() {
+            if !remove_party_indexes.contains(&refresh_message.party_index) {
+                refresh_message.remove_party_indexes = remove_party_indexes.clone();
+            } else {
+                let mut new_remove_party_indexes = remove_party_indexes.clone();
+                new_remove_party_indexes.retain(|value| *value != refresh_message.party_index);
+                refresh_message.remove_party_indexes = new_remove_party_indexes;
+            }
+
+            for (party_index, refresh_bucket) in broadcast_messages.iter_mut() {
+                if refresh_message.remove_party_indexes.contains(party_index) {
+                    continue;
+                }
+                refresh_bucket.push(refresh_message.clone());
+            }
+        }
+
+        for remove_party_index in remove_party_indexes.iter() {
+            assert_eq!(broadcast_messages[remove_party_index].len(), 1);
```

Comment:
what is being checked in this assertion ?

Reply:
In this test, everyone agrees to delete a party (or even multiple ones). I check that the only message that the party has access to it's their own (by default).